### ISSUE TITLE
Error during Java code parsing

### ISSUE THREAD

**Issue by fcav1973** [stoicflame/enunciate#405]
Hi,
I've got an error during in doc module parsing a legal Java XML annotated class. The same class was correctly managed by enunciate 1.28-RC2.
Exception is thrown in:

com.webcohesion.enunciate.modules.jaxb.model.Element line 79:

```
        if ((typeMirror.isArray()) || (typeMirror.isCollection())) {
          throw new EnunciateException("Member " + getName() + " of " + typedef.getQualifiedName() + ": an element choice must not be a collection or an array.");
        }


```
This is the class that generates the error (seems an issue related to _@XmlElement(name = "bytes", type = byte[].class)_ usage):

```
@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
@ThreadSafe
public final class Attribute<T> implements Serializable {

	/**
	 * Warning: Whenever any field of this class is added/removed (or its type
	 * is modified) then serialVersionUID MUST change (increment it). For a
	 * complete list of the serialization incompatible changes, please see
	 * section "5.6.1 Incompatible Changes" at {@link http
	 * ://docs.oracle.com/javase/6/docs
	 * /platform/serialization/spec/version.html}.
	 * 
	 * Please use the same number for the serialVersionUID of this class and of
	 * the SerializationProxy inner class.
	 */
	private static final long serialVersionUID = 1L;

	private String name;

	// maps value types (string, ints, bools etc) + byte[]
	@XmlElements({ @XmlElement(name = "value", type = String.class),
			@XmlElement(name = "bytes", type = byte[].class) })
	@XmlElementWrapper
	private List<T> values;

	// for jaxb
	protected Attribute() {
	}

	public Attribute(final String name, final Collection<T> values) {
		Params.checkStringHasContent("name", name);
		Params.checkValidCharacters("name", name);
		Params.checkNotNull("values", values);
		Params.checkPositive("values.size", values.size());
		this.name = name;
		this.values = Collections.unmodifiableList(new ArrayList<T>(values));
	}

	public Attribute(final String name, final T[] values) {
		this(name, Collections.unmodifiableList(Arrays.asList(values)));
	}

	public Attribute(final String name, final T value) {
		this(name, Collections.singleton(value));
	}

	/**
	 * Attribute name
	 * 
	 * @return attribute name
	 */
	public final String getName() {
		return name;
	}

	/**
	 * Returns the list of all attribute values
	 * 
	 * @return List of attribute values
	 */
	public final List<T> getValues() {
		return values;
	}

	/**
	 * Returns the first value of the attribute
	 * 
	 * @return first value of the attribute
	 */
	public T getValue() {
		if (values.isEmpty()) {
			return null;
		}
		return values.get(0);
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + (name == null ? 0 : name.hashCode());
		result = prime * result + (values == null ? 0 : values.hashCode());
		return result;
	}

	@Override
	public boolean equals(final Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		Attribute<?> other = (Attribute<?>) obj;
		if (name == null) {
			if (other.name != null) {
				return false;
			}
		} else if (!name.equals(other.name)) {
			return false;
		}
		if (values == null) {
			if (other.values != null) {
				return false;
			}
		} else if (!values.equals(other.values)) {
			return false;
		}
		return true;
	}

	@Override
	public String toString() {
		StringBuilder builder = new StringBuilder();
		builder.append("Attribute [name=").append(name).append(", values=")
				.append(values).append("]");
		return builder.toString();
	}

	/**
	 * writeReplace method for the serialization proxy pattern.
	 */
	private Object writeReplace() {
		return new SerializationProxy<T>(this);
	}

	/**
	 * readObject method for the serialization proxy pattern.
	 * 
	 * Note: Thanks to the above writeReplace method, serialization will never
	 * create a serialized instance of the enclosing class, but of the
	 * SerializationProxy inner class. However, an attacker might built it in an
	 * attempt to violate the class’s invariants. Therefore, this readObject
	 * method provides a safe-guard against this case.
	 * 
	 * @param stream
	 * @throws InvalidObjectException
	 */
	private void readObject(final ObjectInputStream stream)
			throws InvalidObjectException {
		throw new InvalidObjectException("SerializationProxy required");
	}

	/**
	 * Serialization proxy for Attribute class.
	 */
	private static class SerializationProxy<T> implements Serializable {

		/**
		 * Warning: Whenever a field the enclosing class is added/removed (or
		 * its type is modified) then serialVersionUID MUST change (increment
		 * it). Please use the same number for the serialVersionUID of enclosing
		 * class and of this SerializationProxy inner class.
		 */
		private static final long serialVersionUID = 1L;

		private final String name;
		private final List<T> values;

		SerializationProxy(final Attribute<T> attribute) {
			this.name = attribute.name;
			this.values = attribute.values;
		}

		private Object readResolve() {
			// Uses public constructor
			return new Attribute<T>(name, values);
		}
	}
}
@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
@ThreadSafe
public final class Attribute<T> implements Serializable {

	/**
	 * Warning: Whenever any field of this class is added/removed (or its type
	 * is modified) then serialVersionUID MUST change (increment it). For a
	 * complete list of the serialization incompatible changes, please see
	 * section "5.6.1 Incompatible Changes" at {@link http
	 * ://docs.oracle.com/javase/6/docs
	 * /platform/serialization/spec/version.html}.
	 * 
	 * Please use the same number for the serialVersionUID of this class and of
	 * the SerializationProxy inner class.
	 */
	private static final long serialVersionUID = 1L;

	private String name;

	// maps value types (string, ints, bools etc) + byte[]
	@XmlElements({ @XmlElement(name = "value", type = String.class),
			@XmlElement(name = "bytes", type = byte[].class) })
	@XmlElementWrapper
	private List<T> values;

	// for jaxb
	protected Attribute() {
	}

	public Attribute(final String name, final Collection<T> values) {
		Params.checkStringHasContent("name", name);
		Params.checkValidCharacters("name", name);
		Params.checkNotNull("values", values);
		Params.checkPositive("values.size", values.size());
		this.name = name;
		this.values = Collections.unmodifiableList(new ArrayList<T>(values));
	}

	public Attribute(final String name, final T[] values) {
		this(name, Collections.unmodifiableList(Arrays.asList(values)));
	}

	public Attribute(final String name, final T value) {
		this(name, Collections.singleton(value));
	}

	/**
	 * Attribute name
	 * 
	 * @return attribute name
	 */
	public final String getName() {
		return name;
	}

	/**
	 * Returns the list of all attribute values
	 * 
	 * @return List of attribute values
	 */
	public final List<T> getValues() {
		return values;
	}

	/**
	 * Returns the first value of the attribute
	 * 
	 * @return first value of the attribute
	 */
	public T getValue() {
		if (values.isEmpty()) {
			return null;
		}
		return values.get(0);
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + (name == null ? 0 : name.hashCode());
		result = prime * result + (values == null ? 0 : values.hashCode());
		return result;
	}

	@Override
	public boolean equals(final Object obj) {
		if (this == obj) {
			return true;
		}
		if (obj == null) {
			return false;
		}
		if (getClass() != obj.getClass()) {
			return false;
		}
		Attribute<?> other = (Attribute<?>) obj;
		if (name == null) {
			if (other.name != null) {
				return false;
			}
		} else if (!name.equals(other.name)) {
			return false;
		}
		if (values == null) {
			if (other.values != null) {
				return false;
			}
		} else if (!values.equals(other.values)) {
			return false;
		}
		return true;
	}

	@Override
	public String toString() {
		StringBuilder builder = new StringBuilder();
		builder.append("Attribute [name=").append(name).append(", values=")
				.append(values).append("]");
		return builder.toString();
	}

	/**
	 * writeReplace method for the serialization proxy pattern.
	 */
	private Object writeReplace() {
		return new SerializationProxy<T>(this);
	}

	/**
	 * readObject method for the serialization proxy pattern.
	 * 
	 * Note: Thanks to the above writeReplace method, serialization will never
	 * create a serialized instance of the enclosing class, but of the
	 * SerializationProxy inner class. However, an attacker might built it in an
	 * attempt to violate the class’s invariants. Therefore, this readObject
	 * method provides a safe-guard against this case.
	 * 
	 * @param stream
	 * @throws InvalidObjectException
	 */
	private void readObject(final ObjectInputStream stream)
			throws InvalidObjectException {
		throw new InvalidObjectException("SerializationProxy required");
	}

	/**
	 * Serialization proxy for Attribute class.
	 */
	private static class SerializationProxy<T> implements Serializable {

		/**
		 * Warning: Whenever a field the enclosing class is added/removed (or
		 * its type is modified) then serialVersionUID MUST change (increment
		 * it). Please use the same number for the serialVersionUID of enclosing
		 * class and of this SerializationProxy inner class.
		 */
		private static final long serialVersionUID = 1L;

		private final String name;
		private final List<T> values;

		SerializationProxy(final Attribute<T> attribute) {
			this.name = attribute.name;
			this.values = attribute.values;
		}

		private Object readResolve() {
			// Uses public constructor
			return new Attribute<T>(name, values);
		}
	}
}


```

**Comment by stoicflame** [comment]
Fixed at 2c5b6b4.

**Comment by stoicflame** [comment]
[Enunciate 2.4 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.4.0).
